### ISSUE TITLE
Polars panics when writing a parquet file with an Azure Storage URI

### ISSUE THREAD

**Issue by dsaad68** [pola-rs/polars#11309]
### Checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://crates.io/crates/polars) of Polars.

### Reproducible example

```rust

use clap::Parser;

use cloud::AzureConfigKey;
use polars::prelude::*;

/*
How to run:
cargo run -- --input data\customers.csv  --output abfss://data@ds22learning22adls.dfs.core.windows.net/output/bronze/customers.parquet
*/

/// Simple program to convet CSV to Parquet
#[derive(Parser, Debug)]
#[command(author, version, about, long_about = None)]
struct Args {
    /// Input file path
    #[arg(short, long, value_name = "INPUT-PATH")]
    input: String,

    /// Output file path
    #[arg(short, long, value_name = "OUTPUT-PATH")]
    output: String,
}

fn main() -> PolarsResult<()> {

    dotenvy::dotenv().unwrap();

    let args = Args::parse();

    let input_path = args.input;
    let output_path = args.output;

    let mut schema = Schema::new();
    schema.with_column("CustomerID".to_string().into(), DataType::Int64);
    schema.with_column("FirstName".to_string().into(), DataType::Utf8);
    schema.with_column("LastName".to_string().into(), DataType::Utf8);
    schema.with_column("FullName".to_string().into(), DataType::Utf8);

    let df_csv = CsvReader::from_path(input_path)?
    .has_header(true)
    .with_dtypes(Some(Arc::new(schema)))
    .with_encoding(CsvEncoding::LossyUtf8)
    .finish()?;

    println!("{}", df_csv);

    let account_name = dotenvy::var("ACCOUNT_NAME").unwrap();
    let access_key = dotenvy::var("ACCESS_KEY").unwrap();
    let sas_key = dotenvy::var("SAS_KEY").unwrap();

    // Propagate the credentials and other cloud options.
    let cloud_options = cloud::CloudOptions::default().with_azure([
        (AzureConfigKey::AccountName, account_name.to_string()),
        (AzureConfigKey::AccessKey, access_key.to_string()),
        (AzureConfigKey::ContainerName, "data".to_string()),
        (AzureConfigKey::SasKey, sas_key.to_string())
        ]);

    let cloud_options = Some(cloud_options);

    df_csv.lazy()
        .sink_parquet_cloud(
            output_path,
            cloud_options,
            Default::default(),
        )
        .unwrap();

    Ok(())
}
```

### Log output

```shell
Finished dev [unoptimized + debuginfo] target(s) in 0.42s
     Running `target\debug\poolstar.exe --input data\customers.csv --output abfss://data@ds0learning0adls.dfs.core.windows.net/output/bronze/customers.parquet`
shape: (635, 4)
┌────────────┬───────────┬─────────────┬───────────────────┐
│ CustomerID ┆ FirstName ┆ LastName    ┆ FullName          │
│ ---        ┆ ---       ┆ ---         ┆ ---               │
│ i64        ┆ str       ┆ str         ┆ str               │
╞════════════╪═══════════╪═════════════╪═══════════════════╡
│ 293        ┆ Catherine ┆ Abel        ┆ Catherine Abel    │
│ 295        ┆ Kim       ┆ Abercrombie ┆ Kim Abercrombie   │
│ 297        ┆ Humberto  ┆ Acevedo     ┆ Humberto Acevedo  │
│ 291        ┆ Gustavo   ┆ Achong      ┆ Gustavo Achong    │
│ …          ┆ …         ┆ …           ┆ …                 │
│ 1987       ┆ Dora      ┆ Verdad      ┆ Dora Verdad       │
│ 1989       ┆ Wanda     ┆ Vernon      ┆ Wanda Vernon      │
│ 1991       ┆ Robert    ┆ Vessa       ┆ Robert Vessa      │
│ 1993       ┆ Caroline  ┆ Vicknair    ┆ Caroline Vicknair │
└────────────┴───────────┴─────────────┴───────────────────┘
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: ComputeError(ErrString("unknown url scheme"))', src\main.rs:68:10
stack backtrace:
   0: std::panicking::begin_panic_handler
             at /rustc/d5c2e9c342b358556da91d61ed4133f6f50fc0c3/library\std\src\panicking.rs:593
   1: core::panicking::panic_fmt
             at /rustc/d5c2e9c342b358556da91d61ed4133f6f50fc0c3/library\core\src\panicking.rs:67
   2: core::result::unwrap_failed
             at /rustc/d5c2e9c342b358556da91d61ed4133f6f50fc0c3/library\core\src\result.rs:1651
   3: enum2$<core::result::Result<tuple$<>,enum2$<polars_error::PolarsError> > >::unwrap<tuple$<>,enum2$<polars_error::PolarsError> >
             at /rustc/d5c2e9c342b358556da91d61ed4133f6f50fc0c3\library\core\src\result.rs:1076
   4: poolstar::main
             at .\src\main.rs:62
   5: core::ops::function::FnOnce::call_once<enum2$<core::result::Result<tuple$<>,enum2$<polars_error::PolarsError> > > (*)(),tuple$<> >
             at /rustc/d5c2e9c342b358556da91d61ed4133f6f50fc0c3\library\core\src\ops\function.rs:250
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
error: process didn't exit successfully: `target\debug\poolstar.exe --input data\customers.csv --output abfss://data@ds0learning0adls.dfs.core.windows.net/output/bronze/customers.parquet` (exit code: 101)
```

### Issue description

I have been attempting on modifying the [write_parquet_cloud](https://github.com/pola-rs/polars/tree/main/examples/write_parquet_cloud) example, which currently writes a parquet file to AWS, in order to make it compatible with Azure Blob Storage. Unfortunately, I've encountered issues as it fails and leads to a panic.

When providing the URI to Azure Data Lake Storage Gen2 format as `abfss://<file_system>@<account_name>.dfs.core.windows.net/<path>/<file_name>`, the program panics.

However, when using the `abfs://<file_system>@<account_name>.dfs.core.windows.net/<path>/<file_name>` format, the program doesn’t panic but fails to write the parquet file to the Azure Blob Storage.

### Expected behavior

I have expected to have the same behavior as [write_parquet_cloud](https://github.com/pola-rs/polars/tree/main/examples/write_parquet_cloud) example.

### Installed versions

<details>
--- rustc version      
rustc 1.72.1 (d5c2e9c34 2023-09-13)

-- cargo.toml
[package]
name = "poolstar"
version = "0.1.0"
edition = "2021"

# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html

[dependencies]
dotenvy = "0.15.7"
chrono = "0.4.31"
clap = { version = "4.4.4" , features = ["derive"] }
polars = { version = "0.33.2", features = ["lazy", "temporal", "describe", "json", "parquet", "dtype-datetime", "polars-io", "cloud", "azure", "cloud_write"] }

</details>

**Comment by ion-elgreco** [comment]
Abfss is for Hadood applications that want to use TLS.

Abfss is also not supported in adlfs, and object store neither

**Comment by dsaad68** [comment]
According to Object Store [docs](https://docs.rs/object_store/latest/object_store/azure/struct.MicrosoftAzureBuilder.html), these urls Schema are supposed:

abfs[s]://<container>/<path> (according to fsspec)
abfs[s]://<file_system>@<account_name>.dfs.core.windows.net/<path>
abfs[s]://<file_system>@<account_name>.dfs.fabric.microsoft.com/<path>
az://<container>/<path> (according to fsspec)
adl://<container>/<path> (according to fsspec)
azure://<container>/<path> (custom)
https://<account>.dfs.core.windows.net
https://<account>.blob.core.windows.net
https://<account>.dfs.fabric.microsoft.com
https://<account>.dfs.fabric.microsoft.com/<container>
https://<account>.blob.fabric.microsoft.com
https://<account>.blob.fabric.microsoft.com/<container>

**Comment by ion-elgreco** [comment]
> According to Object Store [docs](https://docs.rs/object_store/latest/object_store/azure/struct.MicrosoftAzureBuilder.html), these urls Schema are supposed:
> 
> abfs[s]://<container>/<path> (according to fsspec)
> abfs[s]://<file_system>@<account_name>.dfs.core.windows.net/<path>
> abfs[s]://<file_system>@<account_name>.dfs.fabric.microsoft.com/<path>
> az://<container>/<path> (according to fsspec)
> adl://<container>/<path> (according to fsspec)
> azure://<container>/<path> (custom)
> https://<account>.dfs.core.windows.net
> https://<account>.blob.core.windows.net
> https://<account>.dfs.fabric.microsoft.com
> https://<account>.dfs.fabric.microsoft.com/<container>
> https://<account>.blob.fabric.microsoft.com
> https://<account>.blob.fabric.microsoft.com/<container>

It's not in the fsspec protocol list: https://filesystem-spec.readthedocs.io/en/latest/_modules/fsspec/registry.html#available_protocols

**Comment by dsaad68** [comment]
> I think the reason your code is not working with `abfs` is because you need to activate 'streaming' feature.

I have added `streaming` feature and tested it with `abfs` again, it still doesn't write to Azure Storage.

**Comment by Bidek56** [comment]
Your code works fine for me when using `main` branch from local source:

Cargo.toml
```
polars = {path = "/Users/..../git/polars/crates/polars", features = ["streaming", "lazy", "temporal", "describe", "json", "parquet", "dtype-datetime", "polars-io", "azure", "cloud_write"] }

[features]
streaming = []
parquet = []
```

It's possible that something was fixed since `0.33.2`

**Comment by dsaad68** [comment]
> Your code works fine for me when using `main` branch from local source:
> 
> Cargo.toml
> 
> ```
> polars = {path = "/Users/..../git/polars/crates/polars", features = ["streaming", "lazy", "temporal", "describe", "json", "parquet", "dtype-datetime", "polars-io", "azure", "cloud_write"] }
> 
> [features]
> streaming = []
> parquet = []
> ```
> 
> It's possible that something was fixed since `0.33.2`

I have tried it, still the same result.

**Comment by dsaad68** [comment]
> Your code works fine for me when using `main` branch from local source:
> 
> Cargo.toml
> 
> ```
> polars = {path = "/Users/..../git/polars/crates/polars", features = ["streaming", "lazy", "temporal", "describe", "json", "parquet", "dtype-datetime", "polars-io", "azure", "cloud_write"] }
> 
> [features]
> streaming = []
> parquet = []
> ```
> 
> It's possible that something was fixed since `0.33.2`

I have tried this method, and It works now!  👌
What could be the reason?

**Comment by Bidek56** [comment]
It's possible that something was fixed since `0.33.2`
I would wait for the next RS release and try again with the new crate version.
Please close this ticket if your code is working fine. Thx

**Comment by universalmind303** [comment]
it was probably this https://github.com/pola-rs/polars/pull/11240

previously we were using our own mappings for allowed schemes, now we defer it all to object store. 

Closing as it was resolved in #11240
